carmm/build/mixed_oxide.py

```diff
@@ -0,0 +1,98 @@
+import math
+import numpy as np
+from ase import Atoms
+from ase.io import read, write
+import random
+from ase.visualize import view
+
+def mixed_oxide(atoms_object:Atoms, metal1, metal2, ratio, metal1_charge, metal2_charge, metal1_moment, metal2_moment, up_or_down_spin=None):
+    '''
+    Provided a geometry.in file for a metal oxide with formula AxOy, this function creates and saves
+    a geometry_mixed.in file for a mixed oxide with formula Ax-kBkOy where A and B are the two metals.
+    x is the initial stoichiometry of A in AxOy.
+    The value of k can be figured out from the equation ratio(metal_A:metal_B) = x-k/k for a given value of ratio
+    IMPORTANT NOTE: As of now this functionality only works for oxides which the metal A is in only one oxidation state
+    like CoO or Mn2O3, etc.
+
+    :param geometry_file: The geometry.in file for the oxide AxOy with initial moments and charges specified.
+    :param metal1: The metal A. eg: Co, Mn, Fe, etc
+    :param metal2: The metal B in Ax-kBkOy
+    :param ratio: The desired ratio of metal A to metal B in the output geometry. In this case x-k/k
+    :param metal1_charge: The charge on metal A that you have specified in the input geometry.in
+    :param metal2_charge: The charge on metal B that you want in the output geometry_mixed.in.
+    NOTE: make sure that both metal A and B both have same charges (as this may result in a charge imbalance in the
+    final structure
+    :param metal1_moment: The initial moment on metal A that you have specified in the input geometry.in
+    :param metal2_moment:
+    :param up_or_down_spin: if the metal1 has zero magnetic moment and metal2 has a magnetic moment specified by
+    metal2_moment, please specify the sign for the magnetic moment of the metal2. Positive (up-spin) or Negative (down-spin)
+    :return:
+    '''
+    if up_or_down_spin!=None:
+        up_or_down_spin=up_or_down_spin.capitalize()
+        print(up_or_down_spin)
+
+    atoms = atoms_object
+    sym = np.array(atoms.get_chemical_symbols())
+    charge= np.array(atoms.get_initial_charges())
+    mom = np.array(atoms.get_initial_magnetic_moments())
+    mom_sign= np.sign(mom)
+    m1_ind = []
+    for ind,symbol in enumerate(sym):
+        if symbol==metal1.capitalize() and charge[ind]==metal1_charge:
+            if np.abs(mom[ind])==metal1_moment:
```

Comment:
Does this need to be two different if statements when there are no other alternatives?
Surely just merge these into a single if statement with three requirements?